pkg/util/conn_pool.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2020 The Ceph-CSI Authors.
```

Comment:
@nixpanic can we open a connect in https://github.com/ceph/ceph-csi/blob/72e37675945d67d3f88943eedb3fed4d146db42d/pkg/rbd/controllerserver.go#L125 and delete it in https://github.com/ceph/ceph-csi/blob/72e37675945d67d3f88943eedb3fed4d146db42d/pkg/rbd/controllerserver.go#L129

Reply:
We could, and that is probably a good idea. However there are still a few functions that need the credentials so that the `rbd` commandline can be built. Not all places can use the go-ceph API yet.

I'll mark it as a TODO for now, and will work towards removing the passing of the credentials to functions in `rbd_utils.go`.

Reply:
A TODO comment has been added now. @Madhu-1 are you ok with that?

Reply:
@nixpanic ack